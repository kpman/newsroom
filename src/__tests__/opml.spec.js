const opml = require('../opml');

jest.mock('fs');

let fs;

beforeEach(() => {
  fs = require('fs');
  fs.readFileSync = jest.fn();
  fs.readFileSync.mockReturnValue({ toString: () => '<xml></xml>' });
  fs.existsSync = jest.fn();
});

it('should be defined', () => {
  expect(opml).toBeDefined();
});

describe('#opml', () => {
  it('should use absolute file path when file exists', async () => {
    fs.existsSync.mockReturnValue(true);
    const opmlpath = '/User/awesome.opml';
    await opml(opmlpath);
    expect(fs.readFileSync).toBeCalledWith('/User/awesome.opml');
  });

  it('should use relative file path', async () => {
    fs.existsSync.mockReturnValue(false);
    process.cwd = jest.fn(() => '/testonly');
    const opmlpath = './awesome.opml';
    await opml(opmlpath);
    expect(fs.readFileSync).toBeCalledWith('/testonly/awesome.opml');
  });

  it('should parse correctly', async () => {
    const opmlpath = 'awesome.opml';
    const result = await opml(opmlpath);
    expect(result).toMatchSnapshot();
  });

  it('should parse correctly with array', async () => {
    const opmlpath = 'awesome.opml';
    fs.readFileSync.mockReturnValue({
      toString: () => `
        <opml version="2.0">
          <body>
            <outline text="Subscriptions" title="Subscriptions">
              <outline xmlUrl="http://www.theverge.com/rss/index.xml" />
              <outline xmlUrl="https://techcrunch.com/feed/" />
              <outline xmlUrl="http://feeds.mashable.com/Mashable" />
              <outline xmlUrl="http://www.engadget.com/rss.xml" />
            </outline>
          </body>
        </opml>`,
    });
    const result = await opml(opmlpath);
    expect(result).toMatchSnapshot();
  });
});
