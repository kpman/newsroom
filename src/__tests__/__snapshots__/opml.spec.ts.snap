// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`#opml should parse correctly 1`] = `[]`;

exports[`#opml should parse correctly with array 1`] = `
[
  {
    "feedType": undefined,
    "feedUrl": "http://www.theverge.com/rss/index.xml",
    "title": undefined,
    "url": undefined,
  },
  {
    "feedType": undefined,
    "feedUrl": "https://techcrunch.com/feed/",
    "title": undefined,
    "url": undefined,
  },
  {
    "feedType": undefined,
    "feedUrl": "http://feeds.mashable.com/Mashable",
    "title": undefined,
    "url": undefined,
  },
  {
    "feedType": undefined,
    "feedUrl": "http://www.engadget.com/rss.xml",
    "title": undefined,
    "url": undefined,
  },
]
`;
